src/primitives/sphere.jl

```diff
@@ -37,6 +37,36 @@ end
 
 Sphere(p1::Tuple, p2::Tuple, p3::Tuple) = Sphere(Point(p1), Point(p2), Point(p3))
 
+"""
+    Sphere(p1, p2, p3, p4)
+
+A 3D sphere passing through points `p1`, `p2`, `p3` and `p4`.
+"""
+function Sphere(p1::Point{3}, p2::Point{3}, p3::Point{3}, p4::Point{3})
+  v1 = coordinates(p1)
+  v2 = coordinates(p2)
+  v3 = coordinates(p3)
+  v4 = coordinates(p4)
```

Comment:
Let's rename this to x1, x2, x3, x4. We use v1, v2, v3, v4 notation to mean vectors throughout the codebase.